app/services/immunization_service/follow_up.rb

```diff
@@ -86,23 +86,27 @@ def process_vaccine_schedules(vaccine_schedules, client_missed_visits, under_fiv
       vaccine_schedules.each do |vaccine_schedule|
         vaccine_schedule[1].each do |visit|
 
-          missed_antigens = visit[:antigens].select { |antigen| antigen[:can_administer] && antigen[:status] == "pending" && visit[:milestone_status] == "current" }
+          missed_antigens = visit[:antigens].select { |antigen| antigen[:can_administer] && antigen[:status] == "pending" && visit[:milestone_status] == "passed" }
```

Comment:
Seems this is not working because vaccine administration seems not to be saving location_id hence fetch_immunization_clients to able to fetch clients